net/mwan3/files/lib/mwan3/mwan3.sh

```diff
@@ -394,13 +409,14 @@ mwan3_set_general_iptables()
 					-p ipv6-icmp \
 					-m icmp6 --icmpv6-type 137 \
 					-j RETURN
-				# do not mangle outgoing echo request
-				$IPT6 -A mwan3_hook \
-					-m set --match-set mwan3_source_v6 src \
-					-p ipv6-icmp \
-					-m icmp6 --icmpv6-type 128 \
-					-j RETURN
-
+				if [ "$FIX_KERNEL_ACTUAL_SK_BUG" = true ]; then
+					# do not mangle outgoing echo request
+					$IPT6 -A mwan3_hook \
+						-m set --match-set mwan3_source_v6 src \
+						-p ipv6-icmp \
+						-m icmp6 --icmpv6-type 128 \
+						-j RETURN
+				fi
```

Comment:
I don't get why we need to skip mangle here for icmp but not for any other methods like tcp, udp or even for IPv4.
Isn't specifying the outgoing device enough to overcome mangle rules?

Reply:
Yes, but for IPv6 were pinging with source IP rather than by device with #11564, so I added this so that the track pings would go out the correct device.

Reply:
Wouldn't the ping using source IP only be used when the kernel has the bug? Shouldn't this rule also be conditionally added?